### PR TITLE
Always prefer dcbranch over other import groups

### PR DESCRIPTION
**Pull request by shifucun**
This way, dcbranch could be used to block/override bad data.

### CODE REVIEW

File: internal/server/stat/stat_reader.go [hunk 1]
```diff
@@ -211,6 +211,10 @@ func ReadStatsPb(
 			)
 			result[place][sv].PlaceName = obs.PlaceName
 		}
+		// Always prefer data from branch cache.
+		if len(btData) > 0 && btGroup.TableNames()[i] == btGroup.BranchTableName() {
```
Comment (pradh) [inline internal/server/stat/stat_reader.go:215]: To confirm, should we do this per SV / place?  Because what if we fix a certain SV but not others?

Comment (shifucun) [reply internal/server/stat/stat_reader.go]: Updated

File: internal/server/stat/stat_reader.go [hunk 2]
```diff
@@ -192,18 +192,29 @@ func ReadStatsPb(
 		return nil, err
 	}
 	result := map[string]map[string]*pb.ObsTimeSeries{}
+	hasData := map[string]map[string]bool{}
 	for _, p := range places {
 		if _, ok := result[p]; !ok {
 			result[p] = map[string]*pb.ObsTimeSeries{}
+			hasData[p] = map[string]bool{}
```
Comment (pradh) [inline internal/server/stat/stat_reader.go:199]: nit: hasData -> hasBranchData

**Review by pradh — APPROVED** [review]
Thanks!

File: internal/server/stat/stat_reader.go (hunk 2 above)
Comment (shifucun) [reply internal/server/stat/stat_reader.go]: Done

**Review by pradh — APPROVED** [review]
Nice!

**Pull request merged**